gate-core/src/main/java/com/netflix/spinnaker/gate/config/CloudEventHandlerConfiguration.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.config;
+
+import io.cloudevents.spring.mvc.CloudEventHttpMessageConverter;
+import java.util.List;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+public class CloudEventHandlerConfiguration implements WebMvcConfigurer {
+
+  @Override
+  public void configureMessageConverters(List<HttpMessageConverter<?>> converters) {
+    converters.add(0, new CloudEventHttpMessageConverter());
```

Comment:
This is actually a different type, this will help to write an endpoints with CloudEvent input or output,
More information [here](https://cloudevents.github.io/sdk-java/spring.html)

Reply:
What I mean is the `content-type` or `accept` HTTP headers. If this is just `application/json`, then it will override the converter used for every other request in gate.

Reply:
I think this configuration will add to the list of existing converters but not override or replace the configuration.

Reply:
It adds it to the front of the list, though! If you can add it without an index, then that should work, though then you could just define a bean of `CloudEventHttpMessageConverter`.

Reply:
Adding this converter without index is causing an issue with InvalidDefinitionException, 
So keeping index at 0 and defined a bean of `CloudEventHttpMessageConverter`.

Reply:
Putting it at index 0 means making it top priority. That is causing it to be used instead of the JSON message converter you were updating before making the custom one. It might just be simpler to update the properties file like you did previously if this setting is required globally. Alternatively, I'd suggest looking at Jackson's annotations and mixins for a potentially localized solution here, but perhaps that's unnecessary. While Jackson provides numerous knobs for tuning things, it's kind of complicated in my experience.

Reply:
Yes in this case its working only when keeping this converter at index 0.

Updating properties file is actually for objectMapper configuration(SerializationFeature.FAIL_ON_EMPTY_BEANS) and not for CloudEventHttpMessageConverter If I am not wrong.

Yeah I don't see any Jackson's annotations to set the localized configuration at a request level for adding converters like  CloudEventHttpMessageConverter but it is possible when we use rest client for any requests something like below,
`restTemplate.getMessageConverters().add(0, new CloudEventHttpMessageConverter());`

Reply:
Read this: https://www.baeldung.com/spring-httpmessageconverter-rest